c10/cuda/CUDACachingAllocator.h

```diff
@@ -114,6 +130,7 @@ struct SegmentInfo {
   int64_t total_size = 0;
   int64_t allocated_size = 0;
   int64_t active_size = 0;
+  int64_t stream = 0;
```

Comment:
nit: Why not cudaStream_t ?

Reply:
I was just being consistent with how it will get exposed in Python, where I do not want to have to create a real stream object, because they can't be pickled. It's more of an ID for debugging than a handle back to the actual stream.

Reply:
Since it doesn't matter in C++, I switched it so that the conversion to int happens in the Python bindings.